ui/app/adapters/kv/data.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import ApplicationAdapter from '../application';
+import { encodePath } from 'vault/utils/path-encoding-helpers';
+import { kvId } from 'vault/utils/kv-id';
+import { assert } from '@ember/debug';
+
+export default class KvDataAdapter extends ApplicationAdapter {
+  namespace = 'v1';
+
+  _urlForSecret(backend, path, version) {
+    const base = `${this.buildURL()}/${encodePath(backend)}/data/${encodePath(path)}`;
+    return version ? base + `?version=${version}` : base;
+  }
+
+  createRecord(store, type, snapshot) {
+    const { backend, path, version } = snapshot.record;
+    const url = this._urlForSecret(backend, path);
+    return this.ajax(url, 'POST', { data: this.serialize(snapshot) }).then((resp) => {
+      resp.id = kvId(backend, version, path);
+      return resp;
+    });
+  }
+
+  queryRecord(store, type, query) {
+    const { path, backend, version } = query;
+    return this.ajax(this._urlForSecret(backend, path, version), 'GET').then((resp) => {
+      resp.id = kvId(backend, version, path);
+      return resp;
+    });
+  }
+
+  findRecord(store, type, id) {
+    return this.ajax(`${this.buildURL()}/${id}`, 'GET').then((resp) => {
+      resp.id = id;
+      return resp;
+    });
+  }
```

Comment:
I think you're right. But I think I'd need to pick `findRecord` because I believe (asking question in chat) that we want findRecord so it won't fire a network request if the record exists in the store. I'll keep both methods here and wait for Chelsea's feedback.

Reply:
Right now, my though it use queryRecord if you want to make sure you have completely fresh data. And findRecord if you want to avoid when possible making a network request if you already have a cache of the data.